### PR TITLE
Muliti user setup, including many refactors.

### PR DESCRIPTION
**Pull request by cscollett**
I have added multi-user configuration. I also polished it up quite a bit with some recommended refactors through GPT We can go through it in pair programming to see if we want to merge it as-is or revert/rewrite some things.

### CODE REVIEW

File: server/app.py [hunk 1]
```diff
@@ -1,38 +1,45 @@
...
-
 with open(config_path, "r") as config_file:
     config = yaml.safe_load(config_file)
 
-# Specify the file name that will appear in the bucket once uploaded
-object_name = "test.txt"
-
 # URL expiration time in seconds
-url_expiration = 3600  # 1 hour
+URL_EXPIRATION = 3600 # 1 hour
```
Comment (lampham789) [inline server/app.py:19]: We don't want to make URL_EXPIRATION, TOKEN_EXPIRATION, MAX_FILE_NAME_LENGTH configurations?

File: server/app.py [hunk 2]
```diff
@@ -49,48 +56,50 @@ def generate_presigned_url(bucket_name, object_name, expiration):
...
-
-            # encrypted_url = encrypt_message(presigned_url, app.config["public_key"])
-            # return encrypted_url.hex(), 200
-
-            return presigned_url, 200
-        except jwt.JWTError:
-            return "Invalid token", 401
-    else:
-        return "Missing token", 401
+    if not auth_header:
```
Comment (lampham789) [inline server/app.py:62]: I love the guard clauses! This is so much cleaner.

**Review by lampham789 — APPROVED** [review]
Tbh I glossed over the new test files, but I trust that it works. The rest looks good, though. Let's talk about it during pair programming. Also, what is key helper? I thought that was new code you added but I didn't see any definitions in the PR.

File: server/app.py [hunk 3]
```diff
@@ -1,38 +1,45 @@
...
 from pathlib import Path
 
 import boto3
 import yaml
-from cryptography.hazmat.primitives import hashes, serialization
-from cryptography.hazmat.primitives.asymmetric import padding
 from flask import Flask, request
 from jose import jwt
 
+from lib.key_helper import PublicKeyManager
```
Comment (RvanB) [inline server/app.py:9]: Does the lib folder still need to be committed?

Comment (cscollett) [reply server/app.py:9]: Funny that. The big python prepackaged .gitignore excludes lib folders. I updated the .gitignore and now the files are committed.

**Review by lampham789 — APPROVED** [review]

File: server/app.py (hunk 1 above)
Comment (cscollett) [reply server/app.py:19]: I agree, eventually we should build this out, but I was of the mind of keeping it simple while it's a prototype. I even debated with myself removing the JWT token expiration to simplify the client config.

**Review by RvanB — APPROVED** [review]
Looks great!

**Pull request merged**